Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+
+    public static AuditSourceIdentificationType getAuditSourceIdentificationType() {
+        String hcid = HomeCommunityMap.getLocalHomeCommunityId();
+        String homeCommunityName = HomeCommunityMap.getHomeCommunityName(hcid);
+        return AuditDataTransformHelper.createAuditSourceIdentification(hcid, homeCommunityName);
+    }
+
+    protected static AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        // Create Active Participant Section
+        // create a method to call the AuditDataTransformHelper - one expectation
+        AuditMessageType.ActiveParticipant participant = createActiveParticipantFromUser(
+            oUserInfo, Boolean.TRUE);
+        if (null != oUserInfo.getRoleCoded()) {
+            participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(oUserInfo.getRoleCoded().getCode(), "",
+                oUserInfo.getRoleCoded().getCodeSystemName(), oUserInfo.getRoleCoded().getDisplayName()));
+        }
+        return participant;
+    }
+
+    /**
+     * Create the <code>AuditMessageType.ActiveParticipant</code> for an audit log record.
+     *
+     * @param userInfo
+     * @param userIsReq
+     * @return <code>AuditMessageType.ActiveParticipant</code>
+     */
+    private static AuditMessageType.ActiveParticipant createActiveParticipantFromUser(UserType userInfo,
+        Boolean userIsReq) {
+        String ipAddr = null;
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        if (ipAddr == null) {
+            try {
+                ipAddr = InetAddress.getLocalHost().getHostAddress();
+            } catch (UnknownHostException ex) {
+                LOG.error("UnknownHostException thrown getting local host address.", ex);
+                return null;
+            }
+        }
+
+        // Set the User Id
+        String userId = null;
+        if (userInfo != null && userInfo.getUserName() != null && userInfo.getUserName().length() > 0) {
+            userId = userInfo.getUserName();
+        }
+
+        if (userId != null) {
+            participant.setUserID(userId);
+        }
+
+        // If specified, set the User Name
+        String userName = null;
+        if (userInfo != null && userInfo.getPersonName() != null) {
+            if (userInfo.getPersonName().getGivenName() != null && userInfo.getPersonName().getGivenName().length() > 0) {
+                userName = userInfo.getPersonName().getGivenName();
+            }
+
+            if (userInfo.getPersonName().getFamilyName() != null
+                && userInfo.getPersonName().getFamilyName().length() > 0) {
+                if (userName != null) {
+                    userName += (" " + userInfo.getPersonName().getFamilyName());
+                } else {
+                    userName = userInfo.getPersonName().getFamilyName();
+                }
+            }
+        }
+        if (userName != null) {
```

Comment:
same here.  could set this inside a previous if where we know username isn't null.